transactions/scripts/metadata/get_vault_supply_view.cdc

```diff
@@ -0,0 +1,20 @@
+import ExampleToken from "ExampleToken"
+import FungibleTokenMetadataViews from "FungibleTokenMetadataViews"
+import MetadataViews from "MetadataViews"
+
+/// Gets the total supply of the vault's token directly from the vault
+
+pub fun main(address: Address): UFix64 {
+    let account = getAccount(address)
+
+    let vaultRef = account
+        .getCapability(ExampleToken.VaultPublicPath)
+        .borrow<&{MetadataViews.Resolver}>()
+        ?? panic("Could not borrow a reference to the vault resolver")
+
+    let ftSupply = vaultRef.resolveView(Type<FungibleTokenMetadataViews.TotalSupply>())!
```

Comment:
np: Maybe not in scope for this PR, but we could generalize this script if `ExampleToken` implemented `ViewResolver`. I think we should encourage NFT & FT contracts to implement that interface anyway.

Reply:
Why would `ExampleToken` need to implement `ViewResolver` to make this generalizable? Isn't it already generalizable if you just provide the path as an argument instead of hardcoding? Agreed about making it implement `ViewResolver` though.

Reply:
Yeah, totally true. Could just add public path. I've just come across a couple of instances with HC scripts/txns where `ViewResolver` would make things easier, so I've been more inclined to spot opportunities to encourage the pattern. You're right though, either one works